src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java

```diff
@@ -255,8 +255,12 @@ public int read(byte[] b, int off, int len) throws IOException {
             return n;
         }
         // No data in the out buffer, try read new data and decrypt it
-        int nd = decryptMore();
-        if (nd <= 0) {
+        // we loop for new data
+        int nd = 0;
+        while (nd == 0) {
+            nd = decryptMore();
+        }
+        if (nd < 0) {
             return nd;
```

Comment:
@sundapeng 
Hmm...,  it seems OK now. 
It's would be better if we add some testcases in future